### PR TITLE
spelling: Add a github action to point out spelling issues

### PR DESCRIPTION
**Pull request by rgetz**
so they don't need to be fixed just before a release.

### CODE REVIEW

**Comment by rgetz** [comment]
don't merge this yet - github actions don't run on draft PRs,

**Comment by rgetz** [comment]
Thanks - this is Ok to merge now.

**Comment by tfcollins** [comment]
I updated the action rules a bit. Will need approval on all external PRs

File: .github/workflows/codespell.yml
Comment (tfcollins) [inline .github/workflows/codespell.yml:9]: Not sure this is correct. Should at least contain master but [0-9]+.x I don't think applies to our namings

**Comment by pcercuei** [comment]
Does Codespell only parse the changes of a PR, or the whole project every time?

Underlying question - if somebody sends a PR that triggers a false-positive, and I merge it as-is, will the autobuilders be broken in the master branch?

File: .github/workflows/codespell.yml
Comment (rgetz) [reply .github/workflows/codespell.yml]: yeah, this is fixed now.

**Comment by rgetz** [comment]
> I updated the action rules a bit. Will need approval on all external PRs

seems excessive for spell check, but I understand (sort of). 

based on the doc, "By default, all first-time contributors require approval to run workflows." Isn't that good enough?

**Comment by pcercuei** [comment]
I believe if we have it only on pull requests, and one PR gets merged with a typo, then the autobuilder job will fail for every subsequent PR (as it parses the whole project and not just the changes). So I guess the current setup is better.

**Comment by rgetz** [comment]
I guess the other thing is with @tfcollins changes to the way that action rules are enabled - is that an ADI person needs to run the spell check manually for 3rd party PRs. You can merge to master with or without doing a spell check.

**Comment by pcercuei** [comment]
Yes it's fine, I was just wondering if we could do without the `.codespell-ignore.txt` by just ignoring the false positives when merging PRs. But if it affects every subsequent PR then the current system is better.

**Review by tfcollins — APPROVED** [review]

**Pull request merged**
